test/e2e-external/external_db_filestore_test.go

```diff
@@ -50,10 +54,6 @@ func expectValidMattermostInstance(t *testing.T, mattermost *mmv1beta.Mattermost
 
 	err := k8sClient.Create(ctx, mattermost)
 	require.NoError(t, err)
-	defer func() {
-		err = k8sClient.Delete(context.Background(), mattermost)
-		require.NoError(t, err)
-	}()
```

Comment:
I think we still want to cleanup the test instance after the test ends, regardless if successful or not.

Since we expect the instance to exist outside this method, we could pull out creation logic to main test function and call `defer` there.

Reply:
Yes. But figured it out that following code is actually doing the cleanup stuff.

https://github.com/mattermost/mattermost-operator/blob/master/test/e2e-external/main_test.go#L28

And I have verified. Thats why I removed this.

Reply:
Yeah, that cleans up the whole namespace therefore the CR is cleaned up as well. Ok, lets leave it this way then.